### DISCUSSION TITLE
ERC20 contract Approve Function

### DISCUSSION THREAD

**Discussion by mobenet** [smartcontractkit/full-blockchain-solidity-course-py#1038] [Q&A] [answered]
I wanted to know why do we need to use the IERC20.sol interface for the Approve function, if we could use the IWETH.sol Interface, that has also an Approve function? My chance is that ERC20 is the standard and that is why we are using that protocol, but my guess was that maybe the WETH contract (from where we got our weth) could approve our token aswell. Thanks a lot, kind regards <3

**Comment by mobenet** [comment, marked as answer]
I already know the answer, so don't bother! We are using ERC20 contract so that we can approve any type of token based on ERC20. Thanks a lot for your help and for creating this amazing course!

**Reply by cromewar** [reply]
I'm glad you figured out by yourself!
